src/app/components/dashbord-user/new-announcement/new-announcement.component.ts

```diff
@@ -128,80 +199,91 @@ export class NewAnnoucementComponent implements OnInit {
     this.getCitiesByCoutryId(event.countryId);
   }
 
+
   onAnouncementSubmit(){
     if (this.annoucementRequest){
       this.addAnnoucement(); 
     }
       
   }
+
+
+  onUploadImages(event : any) {
+    this.imageUrls=[];
+    this.pictures=event.target.files;
+    console.log("this.pictures first: ",this.pictures)
+    if (event.target.files && event.target.files[0]) {
+      const filesAmount = event.target.files.length;
+      for (let i = 0; i < filesAmount; i++) {
+        const reader = new FileReader();
+        reader.onload = (event: any) => {
+          console.log("event2::",event)
+          this.imageUrls.push(event.target.result);
+        }
+        reader.readAsDataURL(event.target.files[i]);
+      }
+    }
+
+    console.log("this.imageUrls : ",this.imageUrls)
+  }
  
 
   async addAnnoucement() {
     let saveAnnoucement : any
+    console.log("this.annoucementRequest : ",this.annoucementRequest)
     saveAnnoucement = await this.annoucementService.addAnnouncement(this.annoucementRequest);
 
+    console.log("saveAnnoucement : ",saveAnnoucement)
+
     if (saveAnnoucement.announcementId) {
 
      this.uploadFilesWithContext(saveAnnoucement.announcementId)
     }
   }
 
 
-
   showSuccess(msg: string) {
     this.toastr.success('', msg);
   }
 
 
- onUploadImages(event : any) {
-  if (event.target.files && event.target.files[0]) {
-    //this.pictures.push(event.target.files[0], event.target.files[1], event.target.files[2]);
-    this.pictures.push(event.target.files);
-
-    var filesAmount = event.target.files.length;
-    for (let i = 0; i < filesAmount; i++) {
-      var reader = new FileReader();
-      reader.onload = (event: any) => {
-        this.imageUrls.push(event.target.result);
-        
-      }
-      reader.readAsDataURL(event.target.files[i]);
-    }
-  }
-}
-
-
-
  remove_img(i : any){
      this.imageUrls.splice(i, 1)
- 
  }
 
  uploadFilesWithContext(contextId : any){
-  let context = "ANNOUNCEMENT"
   const formData = new FormData();
-  for(let i =0; i<=this.pictures.length; i++){
-    console.log(this.pictures[i])
-    formData.append('file', this.pictures[i]);
-    this.fileService.uploadFilesWithContext(context, contextId, formData).subscribe(
-      (response) => {
-        this.showAlertSuccess = true;
-        this.message = 'Votre annouce a été ajoutée avec succées';
-        this.showSuccess(this.message);
-     
-        
-      },
-      (error) => {
-        console.log(
-          "un erreur a été produit lors de l'actiond e l'announcement ",
-          error
-        );
-      }
-    );
-  }
-//formData.append('file', this.pictures[0]);
-//formData.append('context', 'ANNOUNCEMENT');
+   if (this.pictures) {
+
+     for (let i = 0; i <= this.pictures.length; i++) {
+       console.log("pictures[i] : ",this.pictures[i])
+       formData.append('contextId', contextId)
+       formData.append('file', this.pictures[i]);
```

Comment:
ici, on envoir un fichier par fichier, non? est -il possible d'envoyer touts les fichiers une seule fois?
Dans un autre projet, on a fait comme ca pour envoyer tout les fichiers dans un seul appel:
async savePhotoUser() {
    const formData = new FormData();
    this.files.forEach((file: File) => {
      formData.append('file', file);
    });
    await this.fileService
      .uploadPhotoProject(
        formData,
        this.project.uuid,
        MediaContext[MediaContext.PICTURE_PROJECT]
      )
      .toPromise();
  }

Reply:
c fait